libr/core/cmd.c

```diff
@@ -3199,13 +3205,35 @@ R_API void r_core_cmd_init(RCore *core) {
 	r_cmd_add (core->rcmd, "$",        "alias", &cmd_alias);
 	r_cmd_add (core->rcmd, ".",        "interpret", &cmd_interpret);
 	r_cmd_add (core->rcmd, "/",        "search kw, pattern aes", &cmd_search);
-	cmd_search_init ();
 	r_cmd_add (core->rcmd, "-",        "open cfg.editor and run script", &cmd_stdin);
 	r_cmd_add (core->rcmd, "(",        "macro", &cmd_macro);
 	r_cmd_add (core->rcmd, "u",        "uname/undo", &cmd_uname);
 	r_cmd_add (core->rcmd, "quit",     "exit program session", &cmd_quit);
 	r_cmd_add (core->rcmd, "Q",        "alias for q!", &cmd_Quit);
 	r_cmd_add (core->rcmd, "L",        "manage dynamically loaded plugins", &cmd_plugins);
 
+	cmd_search_init (core);
```

Comment:
no, the best thing would be to accept an array of structs.

typedef struct {
  const char *name;
  const char *desc;
  RCmdCallback cb;
  RCmdInit init;
} RCmdEntry;

RCmdEntry entries[] = {
 { "quit", "exit program", cmd_quit, cmd_exit_init },
 { ...} 
 {NULL}
}

r_cmd_add_all (core->cmd, entries);